### PR TITLE
Embed Resources for backward compatibility

### PR DESCRIPTION
**Pull request by hqhq**
Fixes: https://github.com/docker/docker/issues/19329

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

### CODE REVIEW

**Comment by cyphar** [comment]
Could this cause issues with `Resouces` no longer being a pointer?

File: libcontainer/configs/cgroup_unix.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ type Cgroup struct {
 	ScopePrefix string `json:"scope_prefix"`
 
 	// Resources contains various cgroups settings to apply
-	Resources *Resources `json:"resources"`
+	Resources
```
Comment (LK4D4) [inline libcontainer/configs/cgroup_unix.go]: I sort of don't understand why we need to embed it? Why just don't use value instead of pointer?

Comment (LK4D4) [inline libcontainer/configs/cgroup_unix.go]: Oh, sorry, it's for backward compatibility.

**Comment by mrunalp** [comment]
Hmm, the idea behind making Resources a pointer was that in the future we can clean up the API so if Resources are nil, we can just join instead of create.

**Comment by LK4D4** [comment]
@mrunalp I think we can embed pointer.

**Comment by mrunalp** [comment]
Cool. I would prefer that.

**Comment by hqhq** [comment]
@mrunalp @LK4D4 Updated.

**Comment by crosbymichael** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**